torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -120,44 +123,185 @@ def can_auto_functionalize(op: OperatorBase) -> bool:
     return True
 
 
+@dataclass
+class ViewInfo:
+    base_index: int
+    size: Any = None
+    stride: Any = None
+    storage_offset: Any = None
+    # When is_view is false, the tensor is the base.
+    # When is_view is true, size, stride and storage_offset are all None.
```

Comment:
Shouldn't it be that when is_view is false, the tensor is the base and size stride storage_offset are all None?

Reply:
yes